src/cluster_asm.c

```diff
@@ -2274,8 +2273,9 @@ int slotSnapshotSaveRio(int req, rio *rdb, int *error) {
     if (rioWriteBulkString(rdb, "SNAPSHOT-EOF", 12) == 0) goto werr;
     return C_OK;
 
+werr2:
+    kvstoreResetDictIterator(&kvs_di);
 werr:
```

Comment:
Would there be fewer changes as this way?
```suggestion
werr:
    kvstoreResetDictIterator(&kvs_di);
werr2:
```

Reply:
Yes, I was trying to minimize the changes to existing code when adding the `werr2` label: there're 12 `goto werr;` vs 7 `goto werr2;` here and 5 vs 5 in the rdb.c file below.
I does make sense to have them in `werr`, `werr2` order though.
Let me know if you think we should change it.

Reply:
let's choose the smaller side.